File: Intent Demo/SecondActivity.java
package com.example.demo4;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Toast;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        Intent intent = getIntent();
        String data = ((Intent) intent).getStringExtra("NAME").toString();
        Toast.makeText(this, data, Toast.LENGTH_SHORT).show();

    }
}

File: Intent Demo/SecondActivity.kt
package com.example.demo4

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class SecondActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)
        val data = intent.getStringExtra("NAME") ?: ""
        Toast.makeText(this, data, Toast.LENGTH_SHORT).show()
    }
}
